### PR TITLE
Add support to parse multiple cores for setting CPU affinity

### PR DESCRIPTION
**Pull request by saikishor**
This PR adds a new method that helps to set multiple CPUs for affinity instead of only one and this will help in choosing couple of CPUs to run the controller_manager etc

### CODE REVIEW

**Comment by firesurfer** [comment]
I just saw the PR by chance and was interested as I authored the original one. I am wondering in which (realtime) scenario would you like to prefer a set of cores? From my understanding every time a thread is executed on the different core we suffer from

a) Context switch overhead
b) Additional cache misses

Both brings in an uncertainty which we do not want in RT scenarios. 


The only case I can imagine this could help is modern CPUs with multiple dies inside (e.g. most Ryzen processors). But the OS scheduler will already take care of trying not to move threads between cores on different dies per default.

File: include/realtime_tools/realtime_helpers.hpp [hunk 1]
```diff
@@ -101,6 +115,17 @@ std::pair<bool, std::string> set_thread_affinity(std::thread & thread, int core)
 */
 std::pair<bool, std::string> set_current_thread_affinity(int core);
 
+/**
+ * Configure the current thread affinity - Tell the scheduler to prefer a certain
+ * core for the current thread.
+ * \note The threads created by the calling thread will inherit the affinity.
+ * \param[in] core the cpu numbers of the core. If an empty vector is passed,
```
Comment (firesurfer) [inline include/realtime_tools/realtime_helpers.hpp:122]: I think you missed updating the comments

Comment (saikishor) [reply include/realtime_tools/realtime_helpers.hpp:122]: Well isn't it how it works. Previously it is -1 and now if the vector is empty should behave same

**Comment by saikishor** [comment]
@firesurfer You can set it at the startup as we do in the ros2_control_node or in the configure of controllers and hardware components

File: include/realtime_tools/realtime_helpers.hpp (hunk 1 above)
Comment (firesurfer) [reply include/realtime_tools/realtime_helpers.hpp:122]: Ah sorry I clicked in the wrong line:

> Tell the scheduler to prefer a certain
 * core for the current thread.
 
 This is what I meant. Should probably be: > Tell the scheduler to prefer a certain set of cores for the current thread

**Comment by firesurfer** [comment]
> @firesurfer You can set it at the startup as we do in the ros2_control_node or in the configure of controllers and hardware components

I am aware of that. But I just wanted to understand why it is an advantage in a RT scenario to use the group of cores instead of tying the thread to a single core.

**Comment by saikishor** [comment]
> I am aware of that. But I just wanted to understand why it is an advantage in a RT scenario to use the group of cores instead of tying the thread to a single core.

By default, it uses mostly Core 0 as other. If you can use a set of performance cores, it would be better for the RT application, if your application needs more computational power than one core, then this definitely helps

**Review by christophfroehlich — APPROVED** [review]
LGTM

**Comment by christophfroehlich** [comment]
@mergifyio backport humble

**Pull request merged**